### ISSUE TITLE
2D array operations trigger `NotImplementedError: Legate needs support for more than 3 dimensions`

### ISSUE THREAD

**Issue by piyueh** [nv-legate/legate.numpy#10]
## Problem

When doing a division that requires a 1D denominator to be broadcasted to a 2D array, and when the shape is larger than a certain size, an exception is raised

```python
Traceback (most recent call last):
  File "<removed>/lib/python3.8/site-packages/legion_top.py", line 394, in legion_python_main
    run_path(args[start], run_name='__main__')
  File "<removed>/lib/python3.8/site-packages/legion_top.py", line 193, in run_path
    exec(code, module.__dict__, module.__dict__)
  File "./test1.py", line 14, in <module>
    c = (a[:, 1:] - a[:, :-1]) / (b[1:] - b[:-1])
  File "<removed>/lib/python3.8/site-packages/legate/numpy/array.py", line 776, in __truediv__
    return self.internal_truediv(
  File "<removed>/lib/python3.8/site-packages/legate/numpy/array.py", line 519, in internal_truediv
    return self.perform_binary_op(
  File "<removed>/lib/python3.8/site-packages/legate/numpy/array.py", line 2054, in perform_binary_op
    out._thunk.binary_op(
  File "<removed>/lib/python3.8/site-packages/legate/numpy/deferred.py", line 4876, in binary_op
    ) = self.runtime.compute_broadcast_transform(
  File "<removed>/lib/python3.8/site-packages/legate/numpy/runtime.py", line 2605, in compute_broadcast_transform
    raise NotImplementedError(
NotImplementedError: Legate needs support for more than 3 dimensions
```

## To reproduce

1. step 1: create a test Python script, say, `test.py`. Its content is:
   ```python
   from legate import numpy
   a = numpy.random.random((400, 2001))
   b = numpy.random.random(2001)
   c = (a[:, 1:] - a[:, :-1]) / (b[1:] - b[:-1]) 
   ```
2. step 2: run `test.py` with `legate`. I'm using this one for my test:
   ```shell
   $ legate --cpus 1 ./test.py
   ```

## Other notes

Using different shapes/sizes to generate `a` and `b` seems to also affect the errors. Smaller shapes/sizes do not give any error. For example, `(4, 21)` and `(21,)` for `a` and `b` respectively do not raise any errors.

Also, using the same shape but with different runtime flags may or may not return errors. For example, using `(40, 201)` for `a` and `(201,)` for `b`:
- Using `legate --cpus 0 --omps 1 --ompthreads 1 ./test.py` works fine. No error.
- Using `legate --cpus 0 --omps 1 --ompthreads -ll:okindhack 1 ./test.py` returns the error of `NotImplementedError: Legate needs support for more than 3 dimensions`.

## My workaround

If I explicitly do the broadcasting before the division, everything is fine.

**Comment by piyueh** [comment]
Just to be clear, I know the error says this is something not implemented. But what confuses me is: this calculation only involves 2D array operations and broadcasting from 1D to 2D, why the error message says something about `more than 3 dimensions`?

**Comment by magnatelee** [comment]
@piyueh This should be fixed by #18. Can you pull and try again? You will only need to reinstall Legate NumPy.

**Comment by piyueh** [comment]
@magnatelee thanks for the fix. It's working now!
